octoprint_octorelay/__init__.py

```diff
@@ -200,83 +206,90 @@ def on_shutdown(self):
         self._logger.info("OctoRelay plugin stopped")
         self._logger.info("--------------------------------------------")
 
-    def get_api_commands(self):
-        return {
-            "update": [],
-            "getStatus": [],
-            "listAllStatus":[],
-        }
+    def update_relay(self, index, get_status=False):
+        try:
+            settings = self._settings.get([index], merged=True)
 
-    def on_api_command(self, command, data):
-        self._logger.debug("on_api_command {}, some_parameter is {}".format(command,data))
+            relay_pin = int(settings["relay_pin"])
+            inverted = settings['inverted_output']
+            cmd_on = settings['cmdON']
+            cmd_off = settings['cmdOFF']
 
-        # added api command to get led status
-        if command == "listAllStatus":
             GPIO.setwarnings(False)
-            activeRelays = []
-            for key in self.get_settings_defaults():
-                settings = self.get_settings_defaults()[key]
-                settings.update(self._settings.get([key]))
-                if settings["active"]:
-                    relay_pin = int(settings["relay_pin"])   
-                    inverted = settings['inverted_output']     
-                    GPIO.setup(relay_pin, GPIO.OUT)
-                    relaydata = dict(
-                        id=key,
-                        name=settings["labelText"],
-                        active=inverted != GPIO.input(relay_pin),
-                    )
-                    activeRelays.append(relaydata)
-            return flask.jsonify(activeRelays)
 
-        index = data['pin']
-
-        settings = self.get_settings_defaults()[index]
-        settings.update(self._settings.get([index]))
-
-        relay_pin = int(settings["relay_pin"])
-        inverted = settings['inverted_output']
-        cmdON = settings['cmdON']
-        cmdOFF = settings['cmdOFF']
+            GPIO.setup(relay_pin, GPIO.OUT)
+            # XOR with inverted
+            led_state = inverted != GPIO.input(relay_pin)
 
-        GPIO.setwarnings(False)
+            if get_status:
+                return led_state
 
-        GPIO.setup(relay_pin, GPIO.OUT)
-        # XOR with inverted
-        ledState = inverted != GPIO.input(relay_pin)
-        
-        # added api command to get led status
-        if command == "getStatus":
-            return flask.jsonify(status=ledState)
-            
-        else:
             self._logger.debug("Ocotrelay before pin: {}, inverted: {}, currentState: {}".format(
                 relay_pin,
                 inverted,
-                ledState
+                led_state
```

Comment:
Consider detaching the feature introduction from the refactoring.
That would make your PR much more compact (less changes).
Also, `ledState` is a legacy from a plugin which OctoRelay was based on (see Readme).
The naming might be is subject for changes in another PR.

Reply:
Made an issue for that #94